### PR TITLE
Rework logging support

### PR DESCRIPTION
**Pull request by code-bringer**
#### Logging
Replace LibLog with Microsoft.Extensions.Logging, that means that ZyRabbit classes have to inject ILogger and ILogger<T> classes into constructors in order to log anything instead of using static LogProvider.
By default logging is disabled, but it can be overridden on DI container level, for example see how it is implemented in Samples

#### Dependency Injection
- Simplify interfaces IDependencyRegister and IDependencyResolver
- Significantly rework SimpleDependencyInjectionContainer
- Add support of open generic types
- Rework Autofac and ServiceCollection plugins
- Add unit tests for dependency injections plugins and SimpleDependencyInjectionContainer

#### Removed projects
- ZyRabbit.Compatibility.Legacy, because we do not support of RawRabbit API v1
- ZyRabbit.DependencyInjection.Ninject, because Ninject is abandoned
- ZyRabbit.Enrichers.ZeroFormatter, because ZeroFormatter is abandoned

#### Samples
ASP.NET Sample App
- Upgrade project to latest libraries
- Use ServiceCollection as DI
- Use default Console logging

Console Sample App
- Upgrade project to latest libraries
- Use SimpleDependencyInjectionContainer as DI
- Use Serilog for logging  

Add samples documentation

#### Misc
- Remove all mentions of .NET451 because it is not supported anymore
- Fix logic of ApplicationQueueSuffixPlugin(it was spotted by failed test) 
- Add constructor argument null checks
- Merge a fix from ZyLAB solution for the known multi-threading issue in PublishAcknowledgeMiddleware
- Upgrade performance tests
- Upgrade and fix integration tests
- Update documentation

### Check List
- [x] All test passed.
- [x] Added documentation _(if applicable)_.
- [x] Tests added to ensure functionality.
- [x] Pull Request to `master` branch.

### CODE REVIEW

File: src/ZyRabbit.DependencyInjection.ServiceCollection/ServiceProviderAdapter.cs [hunk 1]
```diff
@@ -9,25 +12,66 @@ public class ServiceProviderAdapter : IDependencyResolver
...
+			{
+				var additionalTypes = additional.Select(a => a.GetType()).ToArray();
+				var additionalParams = new LinkedList<object>(additional);
 
-		public object GetService(Type serviceType, params object[] additional)
-		{
-			additional = additional ?? new object[0];
-			var service = _provider.GetService(serviceType);
-			return service ?? ActivatorUtilities.CreateInstance(_provider, serviceType, additional);
+				var ctor = serviceType.GetConstructors().FirstOrDefault();
```
Comment (mateinavidad) [inline src/ZyRabbit.DependencyInjection.ServiceCollection/ServiceProviderAdapter.cs:30]: Add comment about single constructor restriction.

File: src/ZyRabbit/DependencyInjection/IDependencyRegister.cs [hunk 2]
```diff
@@ -2,22 +2,62 @@
 
 namespace ZyRabbit.DependencyInjection
 {
+	public enum Lifetime
+	{
+		Singelton,
```
Comment (mateinavidad) [inline src/ZyRabbit/DependencyInjection/IDependencyRegister.cs:7]: *Singleton

File: src/ZyRabbit/DependencyInjection/IDependencyRegister.cs [hunk 3]
```diff
@@ -2,22 +2,62 @@
...
+		Transient
```
Comment (mateinavidad) [inline src/ZyRabbit/DependencyInjection/IDependencyRegister.cs:8]: Singleton

File: src/ZyRabbit.DependencyInjection.ServiceCollection/ServiceProviderAdapter.cs (hunk 1 above)
Comment (code-bringer) [reply src/ZyRabbit.DependencyInjection.ServiceCollection/ServiceProviderAdapter.cs:30]: Fixed

**Review by mateinavidad — APPROVED** [review]
Great work! all good from where i stand.

**Pull request merged**